vm/src/frame.rs

```diff
@@ -163,6 +163,7 @@ impl Frame {
     /// Execute a single instruction.
     #[allow(clippy::cognitive_complexity)]
     fn execute_instruction(&self, vm: &VirtualMachine) -> FrameResult {
+        vm.check_signals();
```

Comment:
I think this contraption makes sense, it inserts signals nicely at a defined point of time.

Reply:
Do you think this is a good point?

Reply:
I think so, we would have to wait for builtin functions to finish, but then again, we do not face threading issues.